arm-trafficmanager/2017-03-01/swagger/trafficmanager.json

```diff
@@ -0,0 +1,845 @@
+        },
+        "protocol": {
+          "type": "string",
+          "description": "Gets or sets the protocol (HTTP or HTTPS) used to probe for endpoint health."
+        },
+        "port": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets or sets the TCP port used to probe for endpoint health."
+        },
+        "path": {
+          "type": "string",
+          "description": "Gets or sets the path relative to the endpoint domain name used to probe for endpoint health."
+        }
+      },
+      "description": "Class containing endpoint monitoring settings in a Traffic Manager profile."
+    },
+    "ProfileProperties": {
+      "properties": {
+        "profileStatus": {
+          "type": "string",
+          "description": "Gets or sets the status of the Traffic Manager profile.  Possible values are 'Enabled' and 'Disabled'."
+        },
+        "trafficRoutingMethod": {
+          "type": "string",
+          "description": "Gets or sets the traffic routing method of the Traffic Manager profile.  Possible values are 'Performance', 'Weighted', 'Priority' or 'Geographic'."
+        },
+        "dnsConfig": {
+          "$ref": "#/definitions/DnsConfig",
+          "description": "Gets or sets the DNS settings of the Traffic Manager profile."
+        },
+        "monitorConfig": {
+          "$ref": "#/definitions/MonitorConfig",
+          "description": "Gets or sets the endpoint monitoring settings of the Traffic Manager profile."
+        },
+        "endpoints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Endpoint"
+          },
+          "description": "Gets or sets the list of endpoints in the Traffic Manager profile."
+        }
+      },
+      "description": "Class representing the Traffic Manager profile properties."
+    },
+    "Profile": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ProfileProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Class representing a Traffic Manager profile."
+    },
+    "ProfileListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Profile"
+          },
+          "description": "Gets the list of Traffic manager profiles."
+        }
+      },
+      "description": "The list Traffic Manager profiles operation response."
+    },
+    "TrafficManagerNameAvailability": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The relative name."
+        },
+        "type": {
+          "type": "string",
+          "description": "Traffic Manager profile resource type."
+        },
+        "nameAvailable": {
```

Comment:
It is not recommended to use type: boolean. Instead, the recommended approach is to use string enums with allowed set of values defined. Could you modify it?